app/models/publisher.rb

```diff
@@ -167,37 +153,34 @@ def owner_identifier
     "oauth#google:#{auth_user_id}"
   end
 
+  def domain_normalization_error_status
+    if self.domain_normalization_status
+      case self.domain_normalization_status.to_sym
+      when :exclusion_list_error
+        "#{I18n.t('activerecord.errors.models.publisher.attributes.brave_publisher_id.exclusion_list_error')} #{Rails.application.secrets[:support_email]}"
+      when :api_error_cant_normalize
+        I18n.t("activerecord.errors.models.publisher.attributes.brave_publisher_id.api_error_cant_normalize")
+      when :invalid_uri
+        I18n.t("activerecord.errors.models.publisher.attributes.brave_publisher_id.invalid_uri")
+      end
+    end
+  end
+
   private
 
   def generate_verification_token
     update_attribute(:verification_token, PublisherTokenRequester.new(publisher: self).perform)
   end
 
-  def normalize_inspect_brave_publisher_id
-    normalize_brave_publisher_id
-    inspect_brave_publisher_id unless errors.any?
-  end
-
-  def normalize_brave_publisher_id
-    require "faraday"
-    self.brave_publisher_id = PublisherDomainNormalizer.new(domain: brave_publisher_id).perform
-  rescue PublisherDomainNormalizer::DomainExclusionError
-    errors.add(
-      :brave_publisher_id,
-      "#{I18n.t('activerecord.errors.models.publisher.attributes.brave_publisher_id.exclusion_list_error')} #{Rails.application.secrets[:support_email]}"
-    )
-  rescue PublisherDomainNormalizer::OfflineNormalizationError => e
-    errors.add(:brave_publisher_id, e.message)
-  rescue Faraday::Error
-    errors.add(
-      :brave_publisher_id,
-      I18n.t("activerecord.errors.models.publisher.attributes.brave_publisher_id.api_error_cant_normalize")
-    )
-  rescue URI::InvalidURIError
-    errors.add(
-      :brave_publisher_id,
-      I18n.t("activerecord.errors.models.publisher.attributes.brave_publisher_id.invalid_uri")
-    )
+  def validate_domain_normalization_status
+    error_status = self.domain_normalization_error_status
```

Comment:
i feel like we should be affirmatively checking normalization status. in this case in `domain_normalization_error_status` we're looking for errors and the default case is to validate.
so if we accidentally added an unknown error status, this validation could pass.

Reply:
I'm going to tighten this up by replacing `domain_normalization_status` with `brave_publisher_id_error`. We can check for the presence of the error code and throw if the description can't be found. This should prevent anything from slipping through the cracks (such as an unknown error status).